### PR TITLE
fix: catalog conversion not handling all cases

### LINKED ISSUE #1749: Impossible to migrate the recipe catalog, leading to impossible "restart" of recipes
### Bug description

Since 1.2.x, the versioning of the catalog of recipe should be happening on behalf of the user https://github.com/containers/podman-desktop-extension-ai-lab/pull/1475 

Unfortunately, I had a recipe that was already pulled and it is impossible to restart it.

Screenshot 2024-09-18 at 16 15 55

user-catalog.json:
```{
  "recipes": [],
  "models": [
    {
      "id": "/Users/slemeur/Downloads/Meta-Llama-3-8B-Instruct-Q8_0.gguf",
      "name": "Meta-Llama-3-8B-Instruct-Q8_0",
      "description": "Model imported from /Users/slemeur/Downloads/Meta-Llama-3-8B-Instruct-Q8_0.gguf",
      "hw": "CPU",
      "file": {
        "path": "/Users/slemeur/Downloads",
        "file": "Meta-Llama-3-8B-Instruct-Q8_0.gguf",
        "size": 8540770496,
        "creation": "2024-04-19T09:09:06.922Z"
      },
      "memory": 8540770496
    }
  ],
  "categories": []
}

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #1749

### What does this PR do?

Update the catalog format detection routine as it did not cover all cases

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#1749 

### How to test this PR?

Have an old user catalog

### CODE REVIEW

File: packages/backend/src/utils/catalogUtils.ts [hunk 1]
```diff
@@ -50,7 +50,9 @@ export function sanitize(rawObject: object): ApplicationCatalog {
 }
 
 export function hasCatalogWrongFormat(raw: object): boolean {
```
Comment (axel7083) [inline packages/backend/src/utils/catalogUtils.ts:52]: May we have a unit unit ?

**Review by lstocchi — APPROVED** [review]
LGTM

File: packages/backend/src/utils/catalogUtils.ts [hunk 2]
```diff
@@ -50,7 +50,9 @@ export function sanitize(rawObject: object): ApplicationCatalog {
...
-  return 'recipes' in raw && Array.isArray(raw.recipes) && !!raw.recipes.find(r => 'models' in r);
+  return (
+    !('version' in raw) || ('recipes' in raw && Array.isArray(raw.recipes) && raw.recipes.find(r => 'models' in r))
```
Comment (lstocchi) [inline packages/backend/src/utils/catalogUtils.ts:54]:
```suggestion
    !('version' in raw) || ('recipes' in raw && Array.isArray(raw.recipes) && !!raw.recipes.find(r => 'models' in r))
```
i think it's still better to keep the !! to convert it to boolean explicitly

Comment (axel7083) [reply packages/backend/src/utils/catalogUtils.ts:54]: Or use `includes` to have a boolean

**Review by axel7083 — APPROVED** [review]
LGTM

**Review by benoitf — APPROVED** [review]

**Pull request merged**